client/src/quic_client.rs

```diff
@@ -312,3 +312,88 @@ impl QuicClient {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use {
+        crate::{
+            quic_client::QuicTpuConnection,
+            tpu_connection::{ClientStats, TpuConnection},
+        },
+        crossbeam_channel::unbounded,
+        solana_sdk::{packet::PACKET_DATA_SIZE, quic::QUIC_PORT_OFFSET, signature::Keypair},
+        solana_streamer::quic::spawn_server,
+        std::{
+            collections::HashMap,
+            net::{SocketAddr, UdpSocket},
+            sync::{
+                atomic::{AtomicBool, Ordering},
+                Arc, RwLock,
+            },
+            time::{Duration, Instant},
+        },
+    };
+
+    #[test]
+    fn test_quic_client_multiple_writes() {
+        solana_logger::setup();
+        let s = UdpSocket::bind("127.0.0.1:0").unwrap();
+        let exit = Arc::new(AtomicBool::new(false));
+        let (sender, receiver) = unbounded();
+        let keypair = Keypair::new();
+        let ip = "127.0.0.1".parse().unwrap();
+        let staked_nodes = Arc::new(RwLock::new(HashMap::new()));
+        let t = spawn_server(
+            s.try_clone().unwrap(),
+            &keypair,
+            ip,
+            sender,
+            exit.clone(),
+            1,
+            staked_nodes,
+            10,
+            10,
+        )
+        .unwrap();
+
+        let addr = s.local_addr().unwrap().ip();
+        let port = s.local_addr().unwrap().port() - QUIC_PORT_OFFSET;
```

Comment:
is this critical to the test?  it can introduce CI flake (panic when unwrapping the bind result) since the port may be occupied or in the privileged range

Reply:
Probably... the server needs to be bound to a port to receive the packets sent by the client. Agreed that it's not ideal that we do have this non-deterministic component by having to bind to a port, but to get a realistic test, I do think this is necessary. This is also similar to what is done in the tests for the quic server.